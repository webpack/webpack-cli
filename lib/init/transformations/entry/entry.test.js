"use strict";

const defineTest = require("../../../utils/defineTest");

defineTest(__dirname, "entry", "entry-0", "'index.js'", "init");
defineTest(__dirname, "entry", "entry-0", "\"index.js\"", "init");
defineTest(__dirname, "entry", "entry-0", ["'index.js'", "'app.js'"], "init");
defineTest(
	__dirname,
	"entry",
	"entry-0",
	{
		index: "'index.js'",
		app: "'app.js'"
	},
	"init"
);

defineTest(
	__dirname,
	"entry",
	"entry-0",
	{
		inject: "something",
		app: "'app.js'",
		inject_1: "else"
	},
	"init"
);
defineTest(__dirname, "entry", "entry-0", "() => 'index.js'", "init");
defineTest(
	__dirname,
	"entry",
	"entry-0",
	"() => new Promise((resolve) => resolve(['./app', './router']))",
	"init"
);
defineTest(__dirname, "entry", "entry-0", "entryStringVariable", "init");

defineTest(__dirname, "entry", "entry-0", "'index.js'", "add");
defineTest(__dirname, "entry", "entry-0", ["'index.js'", "'app.js'"], "add");
defineTest(
	__dirname,
	"entry",
	"entry-1",
	{
		index: "'outdex.js'",
		app: "'nap.js'"
	},
	"add"
);

defineTest(
	__dirname,
	"entry",
	"entry-0",
	{
		inject: "something",
		ed: "'eddy.js'",
		inject_1: "else"
	},
	"add"
);
defineTest(__dirname, "entry", "entry-1", "() => 'index.js'", "add");
defineTest(
	__dirname,
	"entry",
	"entry-0",
	"() => new Promise((resolve) => resolve(['./app', './router']))",
	"add"
);
defineTest(__dirname, "entry", "entry-0", "entryStringVariable", "add");
