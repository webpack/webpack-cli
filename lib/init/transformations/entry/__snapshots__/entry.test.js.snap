// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entry transforms correctly using "entry-0" data 1`] = `
"module.exports = {
  entry: 'index.js'
}
"
`;

exports[`entry transforms correctly using "entry-0" data 2`] = `
"module.exports = {
  entry: ['index.js', 'app.js']
}
"
`;

exports[`entry transforms correctly using "entry-0" data 3`] = `
"module.exports = {
  entry: {
    index: 'index.js',
    app: 'app.js'
  }
}
"
`;

exports[`entry transforms correctly using "entry-0" data 4`] = `
"module.exports = {
  entry: {
    something,
    app: 'app.js',
    else
  }
}
"
`;

exports[`entry transforms correctly using "entry-0" data 5`] = `
"module.exports = {
  entry: () => 'index.js'
}
"
`;

exports[`entry transforms correctly using "entry-0" data 6`] = `
"module.exports = {
  entry: () => new Promise((resolve) => resolve(['./app', './router']))
}
"
`;

exports[`entry transforms correctly using "entry-0" data 7`] = `
"module.exports = {
  entry: entryStringVariable
}
"
`;
